drivers/clk/renesas/rzg2l-cpg.c

```diff
@@ -550,27 +597,43 @@ rzg2l_cpg_sd_mux_clk_register(const struct cpg_core_clk *core,
 
 static unsigned long
 rzg2l_cpg_get_foutpostdiv_rate(struct rzg2l_pll5_param *params,
-			       unsigned long rate)
+			       unsigned long rate,
+				   struct rzg2l_cpg_priv *priv)
 {
 	unsigned long foutpostdiv_rate;
+	unsigned int lanes;
+	unsigned int bpp;
 	unsigned long hsclk;
+	u32 dsi_div;
 
-	/*
-	 * Relationship between hsclk and vclk must follow
-	 * vclk * bpp = hsclk * 8 * lanes
-	 * where vclk: video clock (Hz)
-	 *       bpp: video pixel bit depth
-	 *       hsclk: DSI HS Byte clock frequency (Hz)
-	 *       lanes: number of data lanes
-	 * Currently default value:
-	 *		 bpp is 24
-	 *		 lanes is 1
-	 *		 => hsclk = 3 * vclk
-	 * intin = (hsclk * 16 * refdiv * postdiv1 * postdiv2) / EXTAL_FREQ
-	 * fracin = ((hsclk * 16 * refdiv * postdiv1 * postdiv2) % EXTAL_FREQ) << 24) / EXTAL_FREQ
-	 */
-	hsclk = 3 * rate;
+	if(get_dsi_status()){
+		struct platform_device *pdev;
+		struct device_node *np = of_find_node_by_name(NULL, "dsi");
+
+		if (!np) {
+			pr_err("Failed to find 'dsi' node\n");
+		}
+
+		pdev = of_find_device_by_node(np);
+		lanes = rzg2l_mipi_dsi_get_data_lanes(pdev);
+		bpp = rzg2l_mipi_dsi_get_bpp(pdev);
+	} else {
+		/* Default value */
+		lanes = 4;
+		bpp = 24;
```

Comment:
I don't think we should keep this change. Using the device tree to get the status of a specific compatible node is not ideal. We need a better approach.

Refer to: https://github.com/Renesas-SST/linux-rz/blob/dunfell/rz-sbc/drivers/gpu/drm/rcar-du/rcar_du_crtc.c#L315

Reply:
This is the same thing with previous, we are not inside the bridge chain, so I think easiest way to reach that information is through pdev getting from device node. Even we set bpp and lanes in device tree, we still need to reach to that device node. If you have better approach, please give me ideas to fix this

Reply:
As I mentioned above, in this part, if there are any changes to the panel, this code will be able to handle those changes without us needing to modify it.

Reply:
If the get_dsi_status function handles the status of the compatible MIPI-DSI node, do we actually need these lines to recheck the DSI node?

```
struct platform_device *pdev;
struct device_node *np = of_find_node_by_name(NULL, "dsi");

if (!np) {
	pr_err("Failed to find 'dsi' node\n");
}
```

Reply:
> If the get_dsi_status function handles the status of the compatible MIPI-DSI node, do we actually need these lines to recheck the DSI node?
> 
> ```
> struct platform_device *pdev;
> struct device_node *np = of_find_node_by_name(NULL, "dsi");
> 
> if (!np) {
> 	pr_err("Failed to find 'dsi' node\n");
> }
> ```

This is used for getting device node and passing it to get pdev of that node. I just reuse get_dsi_status() which is defined earlier.
So now I need to check the status and get device node at a time and remove the use of get_dsi_status function ? So that I can remove the definition of get_dsi_status